src/main/java/com/frauas/exercisegenerator/services/ExerciseService.java

```diff
@@ -52,11 +56,15 @@ public Exercise getExerciseById(String id) {
 
     public Exercise createExerciseFromDto(CreateExerciseDto exerciseDto) {
         // TODO: Use actual author resolution via login credentials
-        Author author = this.authorRepository.findByName("default");
+        Optional<User> authorOpt = this.userRepository.findByUsername("default");
+        User author;
 
-        if (author == null) {
-            author = Author.builder().name("default").build();
-            author = this.authorRepository.save(author);
+        if (authorOpt.isPresent() == false) {
```

Comment:
Das kann glaube ich gekürzt werden. Ob das funktioniert weiß ich gerade nicht, ist gerade so aus dem Kopf raus:
`User author = this.userRepository.findByUsername("defualt").orElse(() -> {
            User author = User.builder().username("default").build();
           return this.userRepository.save(author);
}`

Reply:
Mir fällt ein, dass wir eigentlich sogar keine User auf diese Art anlegen sollten, weil die Exercises/Sheets dann dem default zugeordnet werden und keinem richtigen User. Die Situation sollte eigentlich eh nicht auftreten, weil es bedeuten würde, dass jemand ohne Account etwas machen würde. Besser wäre eine Exception zu werfen.